File: go/bob/bob.go
// Package bob contains a function for replying to that dirty rascal, Bob.
package bob

import (
	"strings"
	"unicode"
)

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	if remark == "" {
		return "Fine. Be that way!"
	}

	isQuestion := strings.HasSuffix(remark, "?")
	if loudNoises(remark) {
		if isQuestion {
			return "Calm down, I know what I'm doing!"
		}
		return "Whoa, chill out!"
	}
	if isQuestion {
		return "Sure."
	}
	return "Whatever."
}

func loudNoises(remark string) bool {
	if strings.IndexFunc(remark, unicode.IsLetter) == -1 {
		return false
	}
	return strings.ToUpper(remark) == remark
}

File: nim/bob/bob.nim
import strutils, re

proc isLoudNoise(s: string): bool =
  s.contains(re"[A-Z]") and not s.contains(re"[a-z]")

proc hasQMark(s: string): bool =
  s.endsWith(re"\?\s*")

proc hey*(s: string): string =
  if s.isEmptyOrWhitespace():
    return "Fine. Be that way!"
  elif isLoudNoise(s):
    if hasQMark(s):
      return "Calm down, I know what I'm doing!"
    else:
      "Whoa, chill out!"
  elif hasQMark(s):
    return "Sure."
  else:
    return "Whatever."
      
    

    
